### PR TITLE
Add views dropdown and footer actions to the "move to view" dialog

### LINKED ISSUE #10083: Dialog to move a card to another Lovelace dashboard has an inconstant button
### Checklist

- [X] I have updated to the latest available Home Assistant version.
- [X] I have cleared the cache of my browser.
- [X] I have tried a different browser to see if it is related to my browser.

### Describe the issue you are experiencing

When you want to move a card to another Lovelace dashboard you get the following dialog:
Schermafbeelding 2021-09-23 om 15 26 10

The button below changes to the same name as in the dropdown.

### Describe the behavior you expected

I expected two buttons: 'Cancel' and 'Move' instead of the dashboard name.

### Steps to reproduce the issue

1. Edit a dasboard
2. Hit the three dots menu of a card and select `Move to view`

### What version of Home Assistant Core has the issue?

core-2021.9.6

### What was the last working version of Home Assistant Core?

_No response_

### In which browser are you experiencing the issue with?

_No response_

### Which operating system are you using to run this browser?

_No response_

### State of relevant entities

_No response_

### Problem-relevant frontend configuration

_No response_

### Javascript errors shown in your browser console/inspector

_No response_

### Additional information

_No response_

### PR DESCRIPTION
**Pull request by ghost**
## Proposed change

Before:
image

After:
image

Closes #10083

## Type of change

- [ ] Dependency upgrade
- [ ] Bugfix (non-breaking change which fixes an issue)
- [x] New feature (thank you!)
- [ ] Breaking change (fix/feature causing existing functionality to break)
- [ ] Code quality improvements to existing code or addition of tests

## Example configuration

```yaml

```

## Additional information

- This PR fixes or closes issue: fixes #10083 
- This PR is related to issue or discussion:
- Link to documentation pull request:

## Checklist

- [x] The code change is tested and works locally.
- [x] There is no commented out code in this PR.
- [ ] Tests have been added to verify that the new code works.

If user exposed functionality or configuration variables are added/changed:

- [ ] Documentation added/updated for [www.home-assistant.io][docs-repository]

[docs-repository]: https://github.com/home-assistant/home-assistant.io

### CODE REVIEW

File: src/panels/lovelace/editor/select-view/hui-dialog-select-view.ts [hunk 1]
```diff
@@ -94,12 +96,43 @@ export class HuiDialogSelectView extends LitElement {
             </ha-paper-dropdown-menu>`
           : ""}
         ${this._config
-          ? html` <hui-views-list
```
Comment (bramkragten) [inline src/panels/lovelace/editor/select-view/hui-dialog-select-view.ts:97]: This is the only place `hui-views-list` was used, it is no longer used now, so should be removed.

File: src/panels/lovelace/editor/select-view/hui-dialog-select-view.ts [hunk 2]
```diff
@@ -125,9 +158,20 @@ export class HuiDialogSelectView extends LitElement {
     }
   }
 
-  private _selectView(e: CustomEvent): void {
-    const view: number = e.detail.view;
-    this._params!.viewSelectedCallback(this._urlPath!, this._config!, view);
+  private _viewChanged(e: CustomEvent) {
+    const view = Number((e.currentTarget as any).getAttribute("data-index"));
+    if (!isNaN(view)) {
+      this._selectedViewIdx = view;
```
Comment (bramkragten) [inline src/panels/lovelace/editor/select-view/hui-dialog-select-view.ts:164]: `_selectedViewIdx` should be reset when another dashboard is selected

File: src/panels/lovelace/editor/select-view/hui-dialog-select-view.ts [hunk 3]
```diff
@@ -94,12 +96,43 @@ export class HuiDialogSelectView extends LitElement {
...
+                  "ui.panel.lovelace.editor.select_view.views_label"
+                )}
+              >
+                <paper-listbox
+                  slot="dropdown-content"
+                  .selected=${this._selectedViewIdx}
+                >
+                  ${this._config.views.map(
+                    (view, idx) => html`
+                      <paper-icon-item
```
Comment (bramkragten) [inline src/panels/lovelace/editor/select-view/hui-dialog-select-view.ts:111]: We should only reserve space for an icon if any of the views have an icon, check how `hui-views-list` did this

**Comment by bramkragten** [comment]
I'm not a big fan of this UX personally, the old UI could definitely use some clarification, but I feel like having to open another dropdown doesn't really help...

I think the buttons are a good idea, but lets keep a list, and add a selection indicator.

Let's wait for the verdict of @matthiasdebaat on this :-)

**Comment by matthiasdebaat** [comment]
> I think we should keep the UI consistent even if there is only one...
> 
> Just make it checked. Not really a need to code around this case

When there's only one view, there aren't tabs on the lovelace dashboard am I right? In that case it would be easier and more coherent to display a dialog with only a dashboard dropdown. When a users select a dashboard with multiple views it shows the different views.

**Comment by bramkragten** [comment]
> There still is a view. It's just hidden from the header when there is only one.

Yes, but I don't see value in showing that to the user? Just like we dont show to the tabs when there is just 1 view?

**Review by bramkragten — DISMISSED** [review]
Looks good! 1 small remark

**Review by bramkragten — APPROVED** [review]

**Pull request merged**